Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -338,48 +387,84 @@ -(void)updateDisplayOrientedPose:(ARCamera*)camera {
 }
 
 - (void)session:(ARSession *)__unused session didAddAnchors:(nonnull NSArray<__kindof ARAnchor *> *)anchors {
-    if (!planeDetectionEnabled) {
-        return;
+    if (planeDetectionEnabled) {
+        [self LockPlanes];
+        for (ARAnchor* newAnchor : anchors) {
+            if ([newAnchor isKindOfClass:[ARPlaneAnchor class]]) {
+                updatedPlanes.insert((ARPlaneAnchor*)newAnchor);
+            }
+        }
+
+        [self UnlockPlanes];
     }
 
-    [self LockPlanes];
-    for (ARAnchor* newAnchor : anchors) {
-        if ([newAnchor isKindOfClass:[ARPlaneAnchor class]]) {
-            updatedPlanes.insert((ARPlaneAnchor*)newAnchor);
+    if (@available(iOS 13.4, *)) {
```

Comment:
If you do this check where `meshDetectionEnabled` is set (see my other comment), then I don't think you need to check it here (and the caller of `TrySetMeshDetectorEnabled` will see that enabling it failed because it returned false).

Reply:
I think you do need to wrap this block instead though in a `#if (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_13_4)` though since it is again using a type that is only available in newer SDKs.